Bigtable/src/EmulatorSupportTrait.php

```diff
@@ -0,0 +1,50 @@
+<?php
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Bigtable;
+
+use Google\Cloud\Core\EmulatorTrait;
+
+/**
+ * Common method to configure emulator support based on configuration options
+ * and `BIGTABLE_EMULATOR_HOST` environment variable.
+ */
+trait EmulatorSupportTrait
+{
+    use EmulatorTrait;
+
+    /**
+     * Add transport configuration for emulator support if needed.
+     *
+     * @param array $options
+     * @return array
+     */
+    private function setEmulatorOptions(array $options)
```

Comment:
```suggestion
    private function modifyClientOptions(array &$options)
    {
        $emulatorHost = getenv('BIGTABLE_EMULATOR_HOST');

        $options += [
            'hasEmulator' => (bool) $emulatorHost,
            'emulatorHost' => $emulatorHost,
        ];
        if ($options['hasEmulator']) {
            $options += $this->emulatorGapicConfig($options['emulatorHost']);
        }

        return $options;
    }
```

Reply:
Well, if we modify options passed by reference, we don't need `return` anymore.
And `modifyClientOptions` name looks too general to me. Maybe `modifyEmulatorClientOptions`?

Reply:
Yes, you're correct on the first point. I updated the sample. The method name is dictated by the [parent](https://github.com/googleapis/gax-php/blob/master/src/GapicClientTrait.php?rgh-link-date=2020-09-01T18%3A58%3A55Z#L637-L640) method I linked to above. You should document what is going on in the docblock.